google/ggrc-core
# GGRC-5876,GGRC-5807 Update workflow ACL propagation

# Dependencies

This PR is `on hold` until the dependencies are merged:

- [x] https://github.com/google/ggrc-core/pull/8290
- [x] https://github.com/google/ggrc-core/pull/8283
- [x] https://github.com/google/ggrc-core/pull/8265
- [x] https://github.com/google/ggrc-core/pull/8289 - PR has conflicting changes and that ticket has higher priority.


# Issue description

In order to improve our acl propagation, we must first unify how our ACL propagation works. We must ensure that workflows use the same propagation code as the rest of the app.

# Steps to test the changes

Check the propagation rules and compare them with the current propagation code for workflows. The end result for the user should not change at all.

Also do some basic rbac checks and if there are any frontend issues with new propagation (regarding missing buttons or anything like that)

Pay attention for any possible dead code now that a lot of code has been removed.

# Solution description

I have just added proper ACR entries for workflows which ensures that those get propagated by the existing code. I have also removed all workflow propagation specific code that might hide any bugs in the new propagation.

# Sanity checklist




# PR Review checklist
